explainer.md

```diff
@@ -219,78 +226,159 @@ user.
 means that the element and its subtree's paint output (ie the output of the
 layout and paint phase) will not change while the lock is acquired. This, in
 turn, means that the user-agent does not have to finish processing the locked
 element’s subtree when processing the layout and paint phase. In other words,
 the user-agent is free to process parts of the subtree, eliminating them from
-the list of things that have changed (ie the dirty elements list). Note that
+the list of things that have changed (i.e. the dirty elements list). Note that
 this dirty list can be indirectly populated as well by, for example, changing
 CSS properties that affect some elements on the page. There are some edge cases
 to consider here which will be discussed later in this document.
 
-##### Element.getDisplayLock()
+##### Element.acquireDisplayLock()
 
 With display locking, each of the `Element` objects has a new function,
-`getDisplayLock()` which returns an object of type `DisplayLock`. The rest of
-the interactions happen with this object, which is associated with the element
-that returned the object.
-
-##### DisplayLock.acquire()
+`acquireDisplayLock()` which takes a callback to run when the lock is acquired
+and returns a promise which resolves when all of the work to layout, and paint
+the subtree has completed. The callback given to `acquireDisplayLock()` takes
+a DisplayLockContext object, which provides functionality for more sophisticated
+designs.
 
 Acquire performs the following steps:
 
-* It returns a promise, which resolves when the acquire process is complete,
-  then
 * It finishes all of the update phases for the current state
 * It saves the output of the layout and paint phase (the painted draw commands)
-  and stashes them to be used when the locked element needs to be painted.
+  and stashes them to be used when the locked element needs to be painted. Note
+  that in the case of an element that is outside of the DOM, this painted output
+  is empty.
 * It sets a flag on the associated element to lock it for visual updates.
-* It enters the element locked state and resolves the returned promise.
+* It returns a promise which, when resolved or rejected, indicates that the lock
+  has been released.
+  <!-- TODO: define the "next opportunity" time -->
+* Finally, at the next opportunity, it runs the given callback passing it a
+  DisplayLockContext for the locked element.
 
 ##### DOM modifications in the locked state
 
-When the element is in the locked state, modifications to it or its subtree’s
-DOM, or modifications to style that affect it or its subtree perform the
-following steps:
+When the element is in the locked state, the callback can make changes to the
+element's subtree. Specifically changes to DOM, or style that affect its subtree
+update the DOM in such a way that script can inspect it immediately, **with the
+exception of style-inducing properties** (see
```

Comment:
style and layout

Reply:
Also note that style and layout inducing properties, when queried, will force style or layout synchronously,